doc/compiler/runtime/Recompilation.md

```diff
@@ -0,0 +1,142 @@
+<!--
+Copyright (c) 2019, 2019 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Background
+
+A natural extension of a JIT Compiler's ability to compile methods "just 
+in time" is to *re*compile methods "just in time". There are several 
+circumstances in when a method might be recompiled; these are outlined 
+in this doc.
+
+# Reasons for recompilation
+
+This section outlines the various reasons for the
+recompilation of a method. Recompilation can either be optional or 
+mandatory. However, since since any compilation is technically not
+mandatory for functional correctness (it exists for performance
+only), both *optional* and *mandatory* in the previous sentence should
+be suffixed with "in order to continue executing the existing compiled
```

Comment:
that wording is odd - perhaps to in order to continue executing a compiled version of the method or something like that?